src/protocols/secure_channel/CASESession.cpp

```diff
@@ -44,21 +45,31 @@ using namespace Crypto;
 using namespace Credentials;
 using namespace Messaging;
 
-constexpr uint8_t kIPKInfo[] = { 0x49, 0x64, 0x65, 0x6e, 0x74, 0x69, 0x74, 0x79, 0x50, 0x72, 0x6f,
-                                 0x74, 0x65, 0x63, 0x74, 0x69, 0x6f, 0x6e, 0x4b, 0x65, 0x79 };
-
-constexpr uint8_t kKDFSR2Info[]   = { 0x53, 0x69, 0x67, 0x6d, 0x61, 0x52, 0x32 };
-constexpr uint8_t kKDFSR3Info[]   = { 0x53, 0x69, 0x67, 0x6d, 0x61, 0x52, 0x33 };
+constexpr uint8_t kKDFSR2Info[]   = { 0x53, 0x69, 0x67, 0x6d, 0x61, 0x32 };
+constexpr uint8_t kKDFSR3Info[]   = { 0x53, 0x69, 0x67, 0x6d, 0x61, 0x33 };
 constexpr size_t kKDFInfoLength   = sizeof(kKDFSR2Info);
 constexpr uint8_t kKDFSEInfo[]    = { 0x53, 0x65, 0x73, 0x73, 0x69, 0x6f, 0x6e, 0x4b, 0x65, 0x79, 0x73 };
 constexpr size_t kKDFSEInfoLength = sizeof(kKDFSEInfo);
 
-constexpr uint8_t kIVSR2[] = { 0x4e, 0x43, 0x41, 0x53, 0x45, 0x5f, 0x53, 0x69, 0x67, 0x6d, 0x61, 0x52, 0x32 };
-constexpr uint8_t kIVSR3[] = { 0x4e, 0x43, 0x41, 0x53, 0x45, 0x5f, 0x53, 0x69, 0x67, 0x6d, 0x61, 0x52, 0x33 };
-constexpr size_t kIVLength = sizeof(kIVSR2);
+constexpr uint8_t kTBEData2_Nonce[] =
+    /* "NCASE_Sigma2N" */ { 0x4e, 0x43, 0x41, 0x53, 0x45, 0x5f, 0x53, 0x69, 0x67, 0x6d, 0x61, 0x32, 0x4e };
+constexpr uint8_t kTBEData3_Nonce[] =
+    /* "NCASE_Sigma3N" */ { 0x4e, 0x43, 0x41, 0x53, 0x45, 0x5f, 0x53, 0x69, 0x67, 0x6d, 0x61, 0x33, 0x4e };
+constexpr size_t kTBEDataNonceLength = sizeof(kTBEData2_Nonce);
+static_assert(sizeof(kTBEData2_Nonce) == sizeof(kTBEData3_Nonce), "TBEData2_Nonce and TBEData3_Nonce must be same size");
 
+// TODO: move this constant over to src/crypto/CHIPCryptoPAL.h - name it CHIP_CRYPTO_AEAD_MIC_LENGTH_BYTES
 constexpr size_t kTAGSize = 16;
 
+constexpr size_t kDestinationMessageLen = kSigmaParamRandomNumberSize + kP256_PublicKey_Length + sizeof(FabricId) + sizeof(NodeId);
+
+// TODO: Remove this list. Replace it with an actual method to retrieve an IPK list (e.g. from a Crypto Store API)
+static uint8_t sIPKList[][kIPKSize] = {
+    { 0 }, /* Corresponds to the FabricID for the Commissioning Example. All zeros. */
+    { 0x1D, 0x1D, 0x1D, 0x1D, 0x1D, 0x1D, 0x1D, 0x1D, 0x1D, 0x1D, 0x1D, 0x1D, 0x1D, 0x1D, 0x1D,
+      0x1D }, /* Corresponds to the FabricID for the Node01_01 Test Vector */
+};
```

Comment:
Here we are coupling the test data with the implementation. It shows that the implementation has insufficient abstraction to be unit tested without completely changing the impl.

If `CaseSession` had basic protected API to get IPK length get access IPKs, at least in short term, then the test could override it and have this test value in its list. Please add a basic abstraction to access the IPK, even if temporary and living in this class, so that testing does not rely baking in this list.

Reply:
Agreed, working on this.

Reply:
This has been updated now.